Instructions how to run WPT locally (#88)
GoogleChromeLabs/chromium-bidi: README.md

python3 examples/cross-browser.py
```

## WPT

WPT is added as a [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules).
To get run WPT tests:

### Check out WPT and setup

#### 1. Check out WPT

```sh
git submodule init
git submodule update
```

#### 2. Go to the WPT folder

```sh
cd wpt
```

> #### 3. [Set up virtualenv](https://web-platform-tests.org/running-tests/from-local-system.html#system-setup)

Reviewer: Already linked below (is my suggestion):

```suggestion
#### 3. Set up virtualenv
```

> Follow the [*System Setup*](https://web-platform-tests.org/running-tests/index.html) instructions.

Reviewer: ```suggestion
Follow the [*System Setup*](https://web-platform-tests.org/running-tests/from-local-system.html#system-setup) instructions.
```

> #### 4. [Setup `hosts` file](https://web-platform-tests.org/running-tests/from-local-system.html#hosts-file-setup):

Reviewer: ```suggestion
#### 4. Setup `hosts` file
```

> To get the WPT tests running, you need to set up the test domains in your hosts file.

Reviewer: ```suggestion
Follow the [`hosts` File Setup](https://web-platform-tests.org/running-tests/from-local-system.html#hosts-file-setup) instructions.
```

And I think the rest can be dropped or abbreviated a lot, it's copy-pasted.

To get the WPT tests running, you need to set up the test domains in your hosts file.

##### On Linux, macOS or other UNIX-like system

```sh
./wpt make-hosts-file | sudo tee -a /etc/hosts
```

##### And on **Windows**

This must be run in a PowerShell session with Administrator privileges:

```sh
python wpt make-hosts-file | Out-File $env:SystemRoot\System32\drivers\etc\hosts -Encoding ascii -Append
```

If you are behind a proxy, you also need to make sure the domains above are excluded from your proxy lookups.

#### 5. Set the `WPT_BROWSER_PATH` environment variable to a Chrome, Edge or Chromium binary to launch. For example, on macOS:

```sh
export WPT_BROWSER_PATH="/Applications/Google Chrome Canary.app/Contents/MacOS/Google Chrome Canary"
export WPT_BROWSER_PATH="/Applications/Microsoft Edge Canary.app/Contents/MacOS/Microsoft Edge Canary"
export WPT_BROWSER_PATH="/Applications/Chromium.app/Contents/MacOS/Chromium"
```

### Run WPT tests
#### 1. Build ChromeDriver BiDi
```sh
npm run build
```
#### 2. Run
```sh
./wpt/wpt run \
  --webdriver-binary ./runBiDiServer.sh \
  --binary $WPT_BROWSER_PATH \
  --manifest ./wpt/MANIFEST.json \
  --metadata ./wpt-metadata \

>   --log-wptreport ./wpt/wptreport.json \

Reviewer: This bit is only needed if you also add instructions for how to update the expectations. That would be great because otherwise one has to rely on getting the expectations from the GitHub Actions artifact. Mentioning that possibility would be good though, sometimes it's hard to reproduce a CI failure locally and it's easier to just copy the files from CI.

Author: I'm not sure how to re-baseline, so let's add the instructions after it will be done for the first time.

Reviewer: Alright. It's the `./wpt update-expectations --product chrome --manifest MANIFEST.json --metadata ../wpt-metadata wptreport.json` bit from the CI setup but the paths will need tweaking.
